### PR TITLE
dateBox - support first century dates input (T905007)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/core/utils/date_serialization.js [hunk 1]
```diff
@@ -9,39 +9,53 @@ const DATETIME_SERIALIZATION_FORMAT = 'yyyy/MM/dd HH:mm:ss';
...
 
-        result = isNumber(parsedValue) ? new Date(parsedValue) : text;
+function parseDate(text) {
+    const parsedValue = !isDate(text) && Date.parse(text);
+
+    if(!parsedValue && getDateSerializationFormat(text) === DATE_SERIALIZATION_FORMAT) {
+
+        const parts = text.match(DATE_SERIALIZATIN_PATTERN);
+        if(parts) {
+            const newDate = new Date(getDimePart(parts[1]), getDimePart(parts[2]), getDimePart(parts[3]));
```
Comment (ksercs) [inline js/core/utils/date_serialization.js]: Did u mean `getTimePart`?

File: testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js [hunk 2]
```diff
@@ -5217,6 +5217,24 @@ QUnit.module('DateBox number and string value support', {
...
+                value: '3/16/1964',
+                min: new Date(-50, 1, 1),
+                displayFormat: 'shortdate'
+            });
+
+            const $input = $('#dateBox').find(`.${TEXTEDITOR_INPUT_CLASS}`);
+
+            $($input.val('1/1/15')).trigger('change');
+
+            assert.deepEqual($('#dateBox').dxDateBox('option', 'text'), '1/1/15');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js:5234]: I think, we can replace this assert with `assert.ok(true, 'no errors has been raised')` in `finally` block for green device tests

Comment (alexander-kotov-dx) [reply testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js:5234]: i fix it by "pickerType: 'calendar'"

File: js/core/utils/date_serialization.js [hunk 3]
```diff
@@ -9,39 +9,53 @@ const DATETIME_SERIALIZATION_FORMAT = 'yyyy/MM/dd HH:mm:ss';
 
 const ISO8601_PATTERN = /^(\d{4,})(-)?(\d{2})(-)?(\d{2})(?:T(\d{2})(:)?(\d{2})?(:)?(\d{2}(?:\.(\d{1,3})\d*)?)?)?(Z|([+-])(\d{2})(:)?(\d{2})?)?$/;
 const ISO8601_TIME_PATTERN = /^(\d{2}):(\d{2})(:(\d{2}))?$/;
+
 const ISO8601_PATTERN_PARTS = ['', 'yyyy', '', 'MM', '', 'dd', 'THH', '', 'mm', '', 'ss', '.SSS'];
+const DATE_SERIALIZATIN_PATTERN = /^(\d{4})\/(\d{2})\/(\d{2})?$/;
```
Comment (ksercs) [inline js/core/utils/date_serialization.js]: 1. `DATE_SERIALIZATIN_PATTERN` -> `DATE_SERIALIZATI**O**N_PATTERN`
2. Why regexp part for day is optional?

File: js/core/utils/date_serialization.js [hunk 4]
```diff
@@ -9,39 +9,53 @@ const DATETIME_SERIALIZATION_FORMAT = 'yyyy/MM/dd HH:mm:ss';
...
     let result;
-    let parsedValue;
 
     if(isString(text) && !skipISO8601Parsing) {
         result = parseISO8601String(text);
     }
 
-    if(!result) {
-        parsedValue = !isDate(text) && Date.parse(text);
+    return result ? result : parseDate(text);
```
Comment (ksercs) [inline js/core/utils/date_serialization.js]: `return result || parseDate(text)`?

File: js/core/utils/date_serialization.js [hunk 5]
```diff
@@ -9,39 +9,53 @@ const DATETIME_SERIALIZATION_FORMAT = 'yyyy/MM/dd HH:mm:ss';
...
+    return +part || 0;
+}
 
-        result = isNumber(parsedValue) ? new Date(parsedValue) : text;
+function parseDate(text) {
+    const parsedValue = !isDate(text) && Date.parse(text);
+
+    if(!parsedValue && getDateSerializationFormat(text) === DATE_SERIALIZATION_FORMAT) {
+
+        const parts = text.match(DATE_SERIALIZATIN_PATTERN);
```
Comment (ksercs) [inline js/core/utils/date_serialization.js:37]: Can we use `RegExp.test` as more fast method?

**Review by DokaRus — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**